### PR TITLE
Add new C++ function for generic healpix pointing matrix calculation.…

### PR DESCRIPTION
**Pull request by tskisner**
…  Wrap that in C/Python.  Use this function in the OpPointingHpix class.  Other small updates to ctoast.py to enable passing NULL pointers to C functions as well as numpy arrays.

### CODE REVIEW

File: src/libtoast/ctoast.cpp [hunk 1]
```diff
@@ -635,8 +635,24 @@ void ctoast_atm_sim_observe( ctoast_atm_sim * sim, double *t, double *az, double
...
 //--------------------------------------
 
+void ctoast_pointing_healpix_matrix ( ctoast_healpix_pixels * hpix, int nest, 
+    double eps, double cal, char const * mode, size_t n, double const * pdata,
+    double const * hwpang, uint8_t const * flags, int64_t * pixels, 
+    double * weights ) {
 
+    toast::healpix::pixels * hp = reinterpret_cast < toast::healpix::pixels * > ( hpix );
+
+    std::string modestr = std::string(mode);
```
Comment (keskitalo) [inline src/libtoast/ctoast.cpp]: why not just 
`std::string modestr(mode);
`?

Comment (tskisner) [inline src/libtoast/ctoast.cpp]: no reason except silly oversight.  Just fixed it.

File: src/python/tod/pointing.py [hunk 2]
```diff
@@ -98,6 +136,11 @@ def __init__(self, pixels='pixels', weights='weights', nside=64, nest=False, mod
         # initialize the healpix pixels object
         self.hpix = hp.Pixels(self._nside)
 
+        if self._mode == "I":
+            self._nnz = 1
+        else:
+            self._nnz = 3
```
Comment (keskitalo) [inline src/python/tod/pointing.py]: We'll have to revisit this if we are mapping only the differenced time streams or we have spurious components.

Comment (tskisner) [inline src/python/tod/pointing.py]: Sure, this operator currently only supports 2 modes.  I can do an explicit elif here and throw an exception for unsupported modes if you like.

**Pull request merged**